plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -171,9 +171,21 @@ export class ElasticSearchSearchEngine implements SearchEngine {
   }
 
   /**
-   * Create a custom search client from the derived elastic search
-   * configuration. This need not be the same client that the engine uses
-   * internally.
+   * Create a custom search client from the derived search client configuration.
+   * This need not be the same client that the engine uses internally.
+   *
+   * @example Instantiate an instance of an Elasticsearch client.
+   * ```ts
+   * import { isOpenSearchCompatible } from '@backstage/plugin-search-backend-module-elasticsearch';
+   * import { Client } from '@elastic/elasticsearch';
+   *
+   * const client = searchEngine.newClient<Client>(options => {
+   *   if (!isOpenSearchCompatible(options)) {
+   *     return new Client(options);
```

Comment:
If I'm not wrong, the options passed to the new client callback do not necessarily have the same type expected by the custom client. If that is true, could we add a remark comment?

Reply:
Yeah, making that explicit sounds good.

Reply:
See: https://github.com/backstage/backstage/pull/15634/commits/0fcbe29c6b8c564e6c23ab6a19fb38dc9791e622